nemo/collections/tts/models/audio_codec.py

```diff
@@ -490,6 +510,28 @@ def _setup_test_dataloader(cfg):
 
     def setup_training_data(self, cfg):
         self._train_dl = self._setup_train_dataloader(cfg)
+        batch_size = cfg['dataloader_params']['batch_size']
+        # Need to set this because if using an IterableDataset, the length of the dataloader is the total number
+        # of samples rather than the number of batches, and this messes up the tqdm progress bar.
+        # So we set the number of steps manually (to the correct number) to fix this.
```

Comment:
Would it work if we defined the `__len__` method of the dataset as `len(self.mapping) / self.batch_size`?

Alternatively would the wrapper `webdataset.ResizedDataset` help? It looks like it allows you to define dataset length and number of steps to limit to in an epoch as separate values.

Reply:
`len(self.mapping) / self.batch_size` I dont think this would work, as we are facing issue with missing world_size but not batch_size. Or we should have a way to pass global batch size than batch_size per GPU